### PR TITLE
Raise RuntimeError when negative integers is entered in the `pow`

### PR DESCRIPTION
**Pull request by marload**
## Description

RuntimeError raised when a negative interger is entered in the `pow` as in pytorch.

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: syft/frameworks/torch/tensors/interpreters/additive_shared.py [hunk 1]
```diff
@@ -604,6 +604,9 @@ def pow(self, power):
          - Divide power by 2 and multiply base to itself (if the power is even)
          - Decrement power by 1 to make it even and then follow the first step
         """
+        if power < 0:
+            raise RuntimeError("Integers to negative integer powers are not allowed.")
```
Comment (Syzygianinfern0) [inline syft/frameworks/torch/tensors/interpreters/additive_shared.py:608]:
```suggestion
            raise RuntimeError("Negative integer powers are not allowed.")
```
Claiming the base of an exponent to be an integer might not be ideal (although it is that way in an encrypted AST) to a user who might have encrypted mayebe a float.

File: syft/frameworks/torch/tensors/interpreters/precision.py [hunk 2]
```diff
@@ -381,6 +381,8 @@ def pow(self, power):
         Args:
             power (int): the exponent supposed to be an integer > 0
         """
+        if power < 0:
+            raise RuntimeError("Integers to negative integer powers are not allowed.")
```
Comment (Syzygianinfern0) [inline syft/frameworks/torch/tensors/interpreters/precision.py:385]:
```suggestion
            raise RuntimeError("Negative integer powers are not allowed.")
```
Missed this one :sweat_smile:

Comment (marload) [reply syft/frameworks/torch/tensors/interpreters/precision.py]: I applied this 👍

**Comment by Syzygianinfern0** [comment]
@marload Feel free to ping any of us on Slack if you want to have to tests re-run due to a flaky failure.

**Comment by marload** [comment]
The test passed! Thank you for your review :smile:

**Pull request merged**